net/libfab/libfab.c

```diff
@@ -92,90 +118,697 @@ M0_INTERNAL void m0_net_libfab_fini(void)
 	*/
 }
 
-/** Used as m0_net_xprt_ops::xo_dom_init(). */
-static int libfab_dom_init(struct m0_net_xprt *xprt, struct m0_net_domain *dom)
+/**
+ * Used to poll for connection and completion events
+ */
+static void libfab_poller(struct m0_fab__tm *tm)
 {
-	struct m0_fab__dom_param *fab_dom;
-	struct fi_info           *fab_hints;
-	int 			  rc;
+	while (tm->ftm_shutdown == false) {
+		struct fi_cq_entry     comp;
+		struct fi_eq_cm_entry  entry;
+		uint32_t               event;
+		ssize_t                rd;
+		int                    wait_cnt;
+		int                    poll_cnt;
+		void                  *ctx[8];
+		int                    i;
+		int                    rc = 0;
+
+		memset(ctx, 0, sizeof(ctx));
+		wait_cnt = fi_wait(tm->ftm_waitset, -1);
+		
+		if (wait_cnt) {
+			poll_cnt = fi_poll(tm->ftm_pollset, ctx, 
+					   ARRAY_SIZE(ctx));
+			for (i = 0; i < poll_cnt; i++) {
+				rc = fi_cq_read(ctx[i], &comp, 1);
+				if (rc > 0) {
+					/* TODO: Action after 
+					succesful completion */
+				} else if (rc < 0 && rc != -FI_EAGAIN) {
+					if (rc == -FI_EAVAIL) {
+						/* TODO: Decode error */
+					}
+					else {
+						/* TODO: Action if err code 
+						not available */
+					}
+				}
+			}
+
+			if (wait_cnt != poll_cnt) {
+				/* TODO: Check all event queues for FI_CONNREQ
+				events */
+				
+				rd = fi_eq_read(tm->ftm_pep->fep_ep_res.fer_eq,
+						&event, &entry, sizeof(entry), 
+						0);
+				if (rd != sizeof(entry)) {
+					/* TODO: Check for error */
+				}
+				if (event == FI_CONNREQ) {
+					/* TODO: Action on incoming connection
+					request */
+				}
+			}
+		}
+	}
+}
+
+/**
+ * This function will extract the ip addr and port from the given ep str
+ */
+static int libfab_ep_addr_decode(const char *ep_name, char *node, char *port)
+{
+	M0_PRE(ep_name != NULL);
+	node = def_node;
+	port = def_port;
+	return M0_RC(0);
+}
```

Comment:
I know this is a stub function, but still this won't work.
the `node` and `port` are passed by value.
They won't change when function returns.